serverless/src/shared/getFilteredTriples.js

```diff
@@ -1,70 +1,109 @@
+ * 3. Constructs another SPARQL query to fetch details for the retrieved concept URIs.
+ * 4. Sends a second request to the SPARQL endpoint to get the concept details.
+ * 5. Processes and returns the SPARQL query results.
  *
  * @async
  * @function getFilteredTriples
  * @param {Object} params - The parameters for filtering triples.
  * @param {string} [params.conceptScheme] - The concept scheme to filter by.
  * @param {string} [params.pattern] - The pattern to filter triples by.
  * @param {string} params.version - The version of the concept scheme to query (e.g., 'published', 'draft', or a specific version number).
+ * @param {number} params.pageNum - The page number for pagination.
+ * @param {number} params.pageSize - The number of items per page.
  * @returns {Promise<Array>} A promise that resolves to an array of triple objects.
  *                           Each triple object contains 's' (subject), 'p' (predicate), and 'o' (object) properties.
  *
  * @throws {Error} If the HTTP request fails, if the response is not ok,
  *                 or if there's any error during the fetching or processing of the triples.
  *
  * @example
- * // Fetch triples for a specific concept scheme in the published version
+ * // Fetch triples for a specific concept scheme in the published version, page 1 with 10 items per page
  * try {
- *   const triples = await getFilteredTriples({ conceptScheme: 'sciencekeywords', version: 'published' });
+ *   const triples = await getFilteredTriples({
+ *     conceptScheme: 'sciencekeywords',
+ *     version: 'published',
+ *     pageNum: 1,
+ *     pageSize: 10
+ *   });
  *   console.log(triples);
  * } catch (error) {
  *   console.error('Failed to get triples:', error);
  * }
  *
  * @example
- * // Fetch triples matching a pattern in the draft version
+ * // Fetch triples matching a pattern in the draft version, page 2 with 20 items per page
  * try {
- *   const triples = await getFilteredTriples({ pattern: 'EARTH SCIENCE', version: 'draft' });
+ *   const triples = await getFilteredTriples({
+ *     pattern: 'EARTH SCIENCE',
+ *     version: 'draft',
+ *     pageNum: 2,
+ *     pageSize: 20
+ *   });
  *   console.log(triples);
  * } catch (error) {
  *   console.error('Failed to get triples:', error);
  * }
  *
- * @see getTriplesForConceptSchemeOrPatternQuery - Used to construct the SPARQL query.
- * @see sparqlRequest - Used to make the SPARQL query request.
+ * @see sparqlRequest - Used to make the SPARQL query requests.
+ * @see getConceptUrisQuery - Used to generate the SPARQL query for fetching concept URIs.
+ * @see getConceptDetailsQuery - Used to generate the SPARQL query for fetching concept details.
  */
 
-export const getFilteredTriples = async ({ conceptScheme, pattern, version }) => {
+export const getFilteredTriples = async ({
+  conceptScheme, pattern, version, pageNum, pageSize
+}) => {
+  const offset = (pageNum - 1) * pageSize
```

Comment:
I'm good with this here but, should we consider having this moved up to the handler that calls this and then passing offset down to here since that value will be constant for any other functions that implement paging on there?